dpc/src/traits/dpc_components.rs

```diff
@@ -37,22 +45,59 @@ pub trait DPCComponents: 'static + Sized {
 
     type InnerScalarField: PrimeField + PoseidonDefaultParametersField;
     type OuterScalarField: PrimeField;
+    type OuterBaseField: PrimeField;
 
     /// Commitment scheme for account contents. Invoked only over `Self::InnerScalarField`.
-    type AccountCommitment: CommitmentScheme + ToConstraintField<Self::InnerScalarField>;
+    type AccountCommitment: CommitmentScheme<Output = Self::AccountCommitmentOutput>
+        + ToConstraintField<Self::InnerScalarField>;
     type AccountCommitmentGadget: CommitmentGadget<Self::AccountCommitment, Self::InnerScalarField>;
+    type AccountCommitmentOutput: ToConstraintField<Self::InnerScalarField>
+        + Clone
+        + Debug
+        + Default
+        + Eq
+        + Hash
+        + ToBytes
+        + FromBytes
+        + Sync
+        + Send;
 
     /// Encryption scheme for account records. Invoked only over `Self::InnerScalarField`.
     type AccountEncryption: EncryptionScheme + ToConstraintField<Self::InnerScalarField>;
     type AccountEncryptionGadget: EncryptionGadget<Self::AccountEncryption, Self::InnerScalarField>;
 
     /// Signature scheme for delegated compute. Invoked only over `Self::InnerScalarField`.
-    type AccountSignature: SignatureScheme + ToConstraintField<Self::InnerScalarField>;
+    type AccountSignature: SignatureScheme<PublicKey = Self::AccountSignaturePublicKey>
+        + ToConstraintField<Self::InnerScalarField>;
     type AccountSignatureGadget: SignatureGadget<Self::AccountSignature, Self::InnerScalarField>;
+    type AccountSignaturePublicKey: ToConstraintField<Self::InnerScalarField>
```

Comment:
Yes, this is the solution to specify a bound on `AccountSignature::PublicKey` because the most direct solutions are unstable.